### DISCUSSION TITLE
relationship .all filter

### DISCUSSION THREAD

**Discussion by riwu** [sqlalchemy/sqlalchemy#12023] [Usage Questions] [answered]
```
class Item(Base):
    name: Mapped[str]

class User(Base):
    id: Mapped[int]
    items: Mapped[Item] = relationship()
```

I can write a query like `select(User).where(User.items.any(name="abc"))`.  
If I want the condition to match for all user items instead, how do I do it?  
I have tried `User.items.all(name="abc")` but it says `all` is not a valid attribute.

**Comment by CaselIT** [comment, marked as answer]
Hi,

I think you are looking for https://docs.sqlalchemy.org/en/20/orm/internals.html#sqlalchemy.orm.PropComparator.and_
